Installing-Oppia-(Mac-OS;-Python-3).md

```diff
@@ -9,62 +9,81 @@
 
-    ```console
-    $ curl https://bootstrap.pypa.io/pip/2.7/get-pip.py -o get-pip2.py
-    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
-                                   Dload  Upload   Total   Spent    Left  Speed
-     100 1863k  100 1863k    0     0  16.7M      0 --:--:-- --:--:-- --:--:-- 18.1M
+    ```shell
+    $ /usr/bin/arch -x86_64 $SHELL --login
     ```
 
-    * Add the following to the `$PATH` variable on your Mac: `/Library/Frameworks/Python.framework/Versions/2.7/bin`
+    This switches the architecture from Mac M1's *ARM* architecture to the emulated *Intel* architecture for the current session. To verify this, run `arch` in the terminal and you should see `i386` being printed. You will need to switch the architecture to Intel for all Oppia development.
 
-    * Download `plistlib.py`
+    If you use Homebrew to install any Python development dependencies for pyenv (discussed below), you will need to install and use Homebrew in this terminal as well. Note that while you can have a Homebrew installation for Apple Silicon and another for Intel architectures installed simultaneously, pyenv is not smart enough to pick the Intel dependencies if both are present, so we recommend using an Intel installation of Homebrew exclusively.
 
-   ```console
-   $ curl https://raw.githubusercontent.com/python/cpython/2.7/Lib/plistlib.py -o 
-   /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plistlib.py
-   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
-                                 Dload  Upload   Total   Spent    Left  Speed
-   100 15185  100 15185    0     0  44274      0 --:--:-- --:--:-- --:--:-- 45193
-   ```
-    * Now try doing `python --version` on your command-line, and you should see version `2.7` printed.
+3. Follow the installation instructions below in your Rosetta terminal.
 
-2. Ensure that you have [Python 3.8](https://www.python.org/downloads/release/python-3812/) installed (Note: you can check this by running `python3 --version`). If Python 3.8 is not installed, download and run the latest Python 3.8 installer from https://www.python.org/downloads/mac-osx/.
+## Install prerequisites
+
+Oppia relies on a number of programs and third-party libraries. Many of these libraries are downloaded automatically for you when you first run the `start.py` script provided with Oppia. However, there are some things that you will need to do beforehand:
+
+1. Install [version 8+ of the Java Runtime Environment (JRE)](https://www.java.com/en/download/).
 
-3. Download [git](http://git-scm.com/download/mac), then run the package and follow instructions. This allows you to store the source in version control.
+2. Install [direnv](https://direnv.net/).
+
+   After finishing with the official installation instructions, you should also create a `~/.direnvrc` file with the following content:
```

Comment:
Make this a new step

Reply:
I disagree. Right now, we have one bullet point for each dependency, which conceptually makes sense. If we made this a new bullet point, then direnv would get 2 steps unlike the other dependencies. I'll update the text above to make clear that each point is a dependency, not a step

Reply:
I believe it would be more intuitive to list each step as an action a user had to take rather than a dependency (Something like what we have for the Linux installation). Users tend to glance over the steps instead of reading each step completely; this may cause issues with their setup.

Reply:
We also group related steps in the Linux instructions. For example, here's step 8 of the [virtual environment setup](https://github.com/oppia/oppia/wiki/Installing-Oppia-%28Linux%3B-Python-3%29#setup-a-virtual-environment):

> Create a virtual environment for oppia by adding file named .envrc into the parent folder of the oppia repository with this content:
> 
> ```console
> use python 3.8.12
> ```
> 
> Then run this command in the same folder:
> 
> ```console
> $ direnv allow
> ```
> 
> Now whenever you are within the oppia folder, the virtual environment will be active.

I think it's the contributor's responsibility to read the instructions thoroughly. If they can't do that, they're going to have a very hard time with the rest of the development cycle, e.g. reading our more complex wiki pages like on debugging. This is especially relevant since we're in the process of raising the bar to contributing.

The benefit of breaking the installation process into steps is that it creates conceptual chunks to organize the information, which is best served by keeping conceptually-related information together. Splitting the process into too many steps is just as confusing as having too few. In the direnv case, if we split this into 2 steps, then we would have 4 steps: 2 of them would tell the user "here's how to install x" while the other two would describe "one of the two steps required to install x". I think that inconsistency is confusing. Instead, I'd rather have 3 steps, each of which say "here's how to install x".

Reply:
sg